fiftyone/core/models.py

```diff
@@ -1958,6 +1958,48 @@ def embed_all(self, args):
         return np.stack([self.embed(arg) for arg in args])
 
 
+class PromptMixin(object):
+    """Mixin for :class:`Model` classes that can generate prompt embeddings.
+
+    This mixin allows for the possibility that only some instances of a class
+    are capable of generating prompt embeddings, per the value of the
+    :meth:`can_embed_prompts` property.
+    """
+
+    @property
+    def can_embed_prompts(self):
```

Comment:
`embed_prompts` is kind of misleading. It sound like we are saving the prompts on the model... can we use something like `text_search_enabled`

Reply:
I like `embed_prompt()` for consistency with the existing `embed()` method for the images